src/fn_iter.rs

```diff
@@ -0,0 +1,41 @@
+/// An iterator that yields values returned by a function.
+pub struct FnIter<F>(F);
```

Comment:
For future reference, iterator structs should be named after the method that creates them as far as possible. So with the current name this would be `Func`.